### DISCUSSION TITLE
How to ddd an external circle to my chord diagram?

### DISCUSSION THREAD

**Discussion by akograf** [jokergoo/circlize#335] [Q&A]
Hi, 
First at all, thanks for this amazing tool!

I generated a chord diagram showing migration between countries.  Now, I would like to add an external circle with one rectangle for each country (aligned at the same position as the inner circle) showing the proportion of descendants that each introduction (migration) originated.

Below you can see an image of my chord diagram, generated with the following code:

```r
#make circo	
circos.clear()
	par(mar = rep(0, 4), cex=1.0)
	circos.par(start.degree = 90, gap.degree = 1)
	chordDiagram(x = matrix, directional = 1, order = countries.order,
             grid.col = countries.col,
             direction.type = c("arrows", "diffHeight"), link.arr.type = "big.arrow",
             annotationTrack = "grid",
             transparency = 0.25,  annotationTrackHeight = c(0.125, 0.125),
             diffHeight  = -0.05)

	#add in labels and axis
	circos.trackPlotRegion(track.index = 1, panel.fun = function(x, y) {
  	sector.index = get.cell.meta.data("sector.index")
  	circos.text(get.cell.meta.data("xcenter"), 0.5, get.cell.meta.data("sector.index"), facing = "bending.outside", niceFacing = TRUE, cex = labelSize) 
	}, bg.border = NA)
	circos.clear()
```

The dataframe I would like to use to add the external circle has the following format:

```
>,df.DescProp
from,to,propDesc
Argentina,Argentina,0.000000000
Lesser,Antilles,Argentina,0.000000000
Bolivia,Argentina,0.007680101
Brazil,Argentina,0.750333117
Ecuador,Argentina,0.000000000
Lucayan,Archipelago,and,Bermuda,Argentina,0.000000000
Chile,Argentina,0.155376405
Colombia,Argentina,0.015172719
Central,America,Argentina,0.000000000
Greater,Antilles,Argentina,0.017683649
Mexico,Argentina,0.015219589
Paraguay,Argentina,0.038534420
Peru,Argentina,0.000000000
Guyanas,Argentina,0.000000000
Uruguay,Argentina,0.000000000
Venezuela,Argentina,0.000000000
```

Where the propDesc column means the proportion of descendants originated from migration coming from the country in the "from column". In other words, I would like to position bars like in the bar_plot below in a circular format and positioned according to the chord digram. 

Would this be possible?

Thanks!

**Comment by jokergoo** [comment]
In the newest version of circlize, such kind of "barplot" is by default supported. See the following example plot:

<img width="522" alt="image" src="https://user-images.githubusercontent.com/449218/192858229-00847324-7034-4f2f-a339-de33ccd54fb3.png">


Note you need to set `diffHeight` to a positive value.
